core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/commands.scala

```diff
@@ -87,3 +87,10 @@ case class WriteStreamOperationStart(
     foreach_batch: Option[StreamingForeachFunction])
     extends LeafNode
     with Command
+
+case class CreateVariable(name: String, dataType: DataType, defaultExpr: Option[Expression], replace: Boolean)
+    extends LeafNode
+    with Command
+
+case class SetVariable(name: String, dataType: Option[DataType], value: Expression) extends LeafNode with Command
```

Comment:
```suggestion
case class SetVariable(name: String, value: Expression, dataType: Option[DataType] = None) extends LeafNode with Command
```
1. optional args always have to be the last
2. use defaults